sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1319,7 +1447,7 @@ object CollapseProject extends Rule[LogicalPlan] with AliasHelper {
 
   def buildCleanedProjectList(
       upper: Seq[NamedExpression],
-      lower: Seq[NamedExpression]): Seq[NamedExpression] = {
+      lower: Iterable[NamedExpression]): Seq[NamedExpression] = {
```

Comment:
Just a question. Where does this PR hand over `non-Seq` type with this method?

Reply:
When we call it with a `ListBuffer` type `mustInlines` argument at: https://github.com/apache/spark/pull/52238/files#diff-11264d807efa58054cca2d220aae8fba644ee0f0f2a4722c46d52828394846efR1343